analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -141,6 +142,126 @@ public void TryGetAttributeValue_ObjectConversion(object value, Type expectedTyp
+        }
+
+        [TestMethod]
+        public void HasAttributeUsageInherited_InheritedUnSpecified()
+        {
+            var code = $$"""
+                    using System;
+
+                    [AttributeUsage(AttributeTargets.All)]
+                    public class MyAttribute: Attribute
+                    {
+                    }
+
+                    [My]
+                    public class Program
+                    {
+                    }
+                    """;
+            var snippet = new SnippetCompiler(code);
+            var attribute = snippet.GetTypeSymbol("Program").GetAttributes().Single(x => x.HasName("MyAttribute"));
+            var actual = attribute.HasAttributeUsageInherited();
+            actual.Should().Be(true); // The default for Inherited = true
+        }
+
+        [TestMethod]
+        public void HasAttributeUsageInherited_NoUsageAttribute()
+        {
+            var code = $$"""
+                    using System;
+
+                    public class MyAttribute: Attribute
+                    {
+                    }
+
+                    [My]
+                    public class Program
+                    {
+                    }
+                    """;
+            var snippet = new SnippetCompiler(code);
+            var attribute = snippet.GetTypeSymbol("Program").GetAttributes().Single(x => x.HasName("MyAttribute"));
+            var actual = attribute.HasAttributeUsageInherited();
+            actual.Should().Be(true); // The default for Inherited = true
+        }
+
+        [DataTestMethod]
+        [DataRow(true, true)]
+        [DataRow(false, true)] // The "Inherited" flag is not inherited for the AttributeUsage attribute itself. See also the SymbolHelperTest.GetAttributesWithInherited... tests,
+                               // where the reflection behavior of MemberInfo.GetCustomAttributes is also tested.
+        public void HasAttributeUsageInherited_UsageInherited(bool inherited, bool expected)
+        {
+            var code = $$"""
+                    using System;
+
+                    [AttributeUsage(AttributeTargets.All, Inherited = {{(inherited ? "true" : "false")}})]
+                    public class BaseAttribute: Attribute
+                    {
+                    }
+
+                    public class MyAttribute: BaseAttribute
+                    {
+                    }
+
+                    [My]
+                    public class Program
+                    {
+                    }
+                    """;
+            var snippet = new SnippetCompiler(code);
+            var attribute = snippet.GetTypeSymbol("Program").GetAttributes().Single(x => x.HasName("MyAttribute"));
+            var actual = attribute.HasAttributeUsageInherited();
+            actual.Should().Be(expected);
+        }
+
+        [TestMethod]
+        public void HasAttributeUsageInherited_DuplicateAttributeUsage()
+        {
+            var code = $$"""
```

Comment:
Should not need interpolation
```suggestion
            var code = """
```

Reply:
Same apply on multiple places in this file.